Sources/HaCWebsiteLib/Controllers/EventApiController.swift

```diff
@@ -0,0 +1,103 @@
+import Kitura
+import KituraNet
+import SwiftyJSON
+import Foundation
+import DotEnv
+
+struct EventApiController {
+  static var handler: RouterHandler = { request, response, next in
+    response.headers["Content-Type"] = "text/plain; charset=utf-8"
+    guard let parsedBody = request.body else {
+      next()
+      response.statusCode = HTTPStatusCode.internalServerError
+      try response.send("Sorry - we weren't able to parse the body of the request\n").end()
+      return
+    }
+    if case .json(let json) = parsedBody {
+      do {
+        try verifyAndSaveEvent(json: json)
+        try response.send("Successfully added your event to the database\n").end()
+      }
+      catch EventParsingError.incorrectPassword {
+        response.statusCode = HTTPStatusCode.unauthorized
+        try response.send("Sorry - your password is incorrect\n").end()
+      } catch EventParsingError.missingParameters {
+        response.statusCode = HTTPStatusCode.badRequest
+        try response.send("Sorry - looks like you didn't include all the necessary fields\n").end()
+      } 
+    } else {
+      response.statusCode = HTTPStatusCode.badRequest
+      try response.send("Please use JSON for post data\n").end()
+    }
+    next()
+  }
+
+  static func verifyAndSaveEvent(json: JSON) throws {
+    if !isCorrectPassword(json: json) {
+      throw EventParsingError.incorrectPassword
+    }
+    let event = try parseEvent(json: json)
+    try event.save()
+  }
+
+  static func isCorrectPassword(json: JSON) -> Bool {
```

Comment:
Not a blocker but conventionally information like passwords go in request headers, not in the body.

Would it make more sense to move this into a middleware so API endpoints automatically get this functionality?

Reply:
Looks like we can just use [this](https://github.com/IBM-Swift/Kitura-CredentialsHTTP) so I'll look into that